server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/MailboxFactory.scala

```diff
@@ -0,0 +1,123 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model
+
+import javax.inject.Inject
+import org.apache.james.jmap.mail.MailboxName.MailboxName
+import org.apache.james.jmap.mail._
+import org.apache.james.jmap.utils.quotas.QuotaLoader
+import org.apache.james.mailbox.model.MailboxACL.EntryKey
+import org.apache.james.mailbox.model.{MailboxCounters, MailboxId, MailboxMetaData, MailboxPath, MailboxACL => JavaMailboxACL}
+import org.apache.james.mailbox.{MailboxSession, Role, SubscriptionManager}
+
+import scala.jdk.CollectionConverters._
+import scala.jdk.OptionConverters._
+
+sealed trait MailboxConstructionOrder
+
+class Factory
+
+class MailboxFactory @Inject() (subscriptionManager: SubscriptionManager) {
+
+  def create(mailboxMetaData: MailboxMetaData,
+             mailboxSession: MailboxSession,
+             allMailboxesMetadata: Seq[MailboxMetaData],
+             quotaLoader: QuotaLoader): Mailbox = {
+
+    val id: MailboxId = mailboxMetaData.getId
+
+    val name: MailboxName = MailboxName.liftOrThrow(mailboxMetaData.getPath
```

Comment:
it's a place where you shouldn't use that method.
This method is meant to be used for cases where a value should not be wrong.
In this case, I doubt you know that because AFAIU you just have a string comment from a split (why it's not a method we can call on the result of getPath BTW?)
This `MailboxFactory.create` method, to respect scala way of writing things should return an Either[String, Mailbox] or even a Validation[Mailbox] and you should compose all the Either generated in this method to return a error about what's wrong in the input.

Reply:
I really start to think `Refined` makes potentially things more complex than easier TBH...

Reply:
What is "more complex" is that you are forced to deal with errors where in java we just wait for exception to throw and show in logs before taking care.

Reply:
I tried to do something in my last commit... Not finished yet but would like some early feedback if possible, I'm not very sure of the way I'm going. Thanks !